### DISCUSSION TITLE
Rename defaultPageParam to initialPageParam

### DISCUSSION THREAD

**Discussion by Moshyfawn** [TanStack/query#5885] [Ideas]
How do you feel about renaming the new Infinite Queries `defaultPageParam` argument to `initialPageParam`?

The use of the word "default" implies the value can be used as **a fallback** when in case of page parameters, it "stands" in line with the rest of the page parameters: it can be used by explicitly "selecting" it via the Infinite Queries methods like `fetchNextPage` or `fetchPreviousPage`; it's not being used in the context of "reset to default page" type of method.

Furthermore, "initialPageParam" name is inline with the Query API argument `initialData` _(and `initialDataUpdatedAt`)_ and the way it's been logically handled by queries.

Let me know what you think. I can already provide a PR since I had to dig through the source code to justify my proposal anyway 😊

**Comment by TkDodo** [comment]
I like it. Please make a PR to `beta` ❤️

**Reply by Moshyfawn** [reply]
#5888
